### ISSUE TITLE
da3_streaming.py Floating point exception

### ISSUE THREAD

**Issue by cChrister** [ByteDance-Seed/Depth-Anything-3#230]
```
~/ccode/Depth-Anything-3/da3_streaming$ python da3_streaming.py --image_dir ./input50
Sim3solve of C++ Version failed, Will using Python Version.
[WARN ] Dependency `gsplat` is required for rendering 3DGS. Install via: pip install git+https://github.com/nerfstudio-project/gsplat.git@0b4dddf04cb687367602c01196913cde6a743d70
The exp will be saved under dir: ./exps/._input50/2026-02-25-11-12-44
config yaml file has been copied to: ./exps/._input50/2026-02-25-11-12-44/base_config.yaml
Loading model...
[INFO ] using SwiGLU layer as FFN
[INFO ] using MLP layer as FFN
Using cache found in /home/chenxiang/.cache/torch/hub/facebookresearch_dinov2_main
Model loaded: ./weights/dino_salad.ckpt
init done.
Loading images from ./input50...
Found 5 images
Processing 5 images in 1                 chunks of size 120 with 60 overlap
[Progress]: 0/1
Loaded 5 images
[INFO ] Processed Images Done taking 0.2578616142272949 seconds. Shape:  torch.Size([5, 3, 280, 504])
[INFO ] Selecting reference view using strategy: middle
*** Aborted at 1771989188 (unix time) try "date -d @1771989188" if you are using GNU date ***
PC: @     0x7857cd76f137 (/home/chenxiang/tool/miniconda3/envs/da3/lib/python3.10/site-packages/nvidia/cublas/lib/libcublas.so.12+0x16f136)
*** SIGFPE (@0x7857cd76f137) received by PID 297926 (TID 0x78587c319740) from PID 18446744072861708599; stack trace: ***
    @     0x78587c0a1ed3 (/usr/lib/x86_64-linux-gnu/libc.so.6+0xa1ed2)
    @     0x7855e4626868 (/home/chenxiang/tool/miniconda3/envs/da3/lib/python3.10/site-packages/pycolmap/_core.cpython-310-x86_64-linux-gnu.so+0xa26867)
    @     0x78587c045330 (/usr/lib/x86_64-linux-gnu/libc.so.6+0x4532f)
    @     0x7857cd76f137 (/home/chenxiang/tool/miniconda3/envs/da3/lib/python3.10/site-packages/nvidia/cublas/lib/libcublas.so.12+0x16f136)
    @     0x7857cd87e4b3 (/home/chenxiang/tool/miniconda3/envs/da3/lib/python3.10/site-packages/nvidia/cublas/lib/libcublas.so.12+0x27e4b2)
    @     0x7857cd87efec (/home/chenxiang/tool/miniconda3/envs/da3/lib/python3.10/site-packages/nvidia/cublas/lib/libcublas.so.12+0x27efeb)
    @     0x7857cd7293ae cublasSdot_v2
    @     0x7856c687337f void at::cuda::blas::dot<float>(cublasContext*, int, float const*, int, float const*, int, float*)
    @     0x7856c6916f9a at::native::dot_cuda(at::Tensor const&, at::Tensor const&) [clone .localalias]
    @     0x7856c656c890 c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoFunctor_<c10::CompileTimeFunctionPointer<at::Tensor (at::Tensor const&, at::Tensor const&), &at::(anonymous namespace)::(anonymous namespace)::wrapper_CUDA__dot>, at::Tensor, c10::g@
    @     0x785703145481 at::_ops::dot::call(at::Tensor const&, at::Tensor const&)
    @     0x7857020c7d33 at::autocast::WrapFunction_<(at::autocast::CastPolicy)4, (c10::DeviceType)1, at::Tensor (at::Tensor const&, at::Tensor const&), &at::_ops::dot::call, at::Tensor, c10::guts::typelist::typelist<at::Tensor const&, at::Tensor const&> >::call(at::Tensor const&,@
    @     0x785701f94880 c10::impl::wrap_kernel_functor_unboxed_<c10::impl::detail::WrapFunctionIntoRuntimeFunctor_<at::Tensor (*)(at::Tensor const&, at::Tensor const&), at::Tensor, c10::guts::typelist::typelist<at::Tensor const&, at::Tensor const&> >, at::Tensor (at::Tensor const@
    @     0x785703145481 at::_ops::dot::call(at::Tensor const&, at::Tensor const&)
    @     0x78571614a6d2 torch::autograd::THPVariable_dot(_object*, _object*, _object*)
    @           0x4fe087 cfunction_call
    @           0x4f77db _PyObject_MakeTpCall.localalias
    @           0x4f3896 _PyEval_EvalFrameDefault
    @           0x4fe4cf _PyFunction_Vectorcall
    @           0x4ee7cf _PyEval_EvalFrameDefault
    @           0x509dde method_vectorcall
    @           0x4f2fd6 _PyEval_EvalFrameDefault
    @           0x509f36 method_vectorcall
    @           0x4f1029 _PyEval_EvalFrameDefault
    @           0x509f36 method_vectorcall
    @           0x4f1029 _PyEval_EvalFrameDefault
    @           0x4f6b2d _PyObject_FastCallDictTstate.localalias
    @           0x5083d0 _PyObject_Call_Prepend
    @           0x5d0c83 slot_tp_call
    @           0x4f77db _PyObject_MakeTpCall.localalias
    @           0x4f3896 _PyEval_EvalFrameDefault
    @           0x4fe4cf _PyFunction_Vectorcall
Floating point exception (core dumped)
```

**Comment by cChrister** [comment]
pip list |grep torch
torch                     2.10.0
torchvision               0.25.0


python -c "import torch; print(f'Torch version: {torch.__version__}'); print(f'CUDA version: {torch.version.cuda}'); print(f'CuDNN version: {torch.backends.cudnn.version()}')"
Torch version: 2.10.0+cu128
CUDA version: 12.8
CuDNN version: 91002

nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Feb__7_19:32:13_PST_2023
Cuda compilation tools, release 12.1, V12.1.66
Build cuda_12.1.r12.1/compiler.32415258_0

**Comment by cChrister** [comment]
correctly match cuda and pytorch cuda version can solve this problem.

nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Feb__7_19:32:13_PST_2023
Cuda compilation tools, release 12.1, V12.1.66
Build cuda_12.1.r12.1/compiler.32415258_0

python -c "import torch; print(torch.version.cuda)"
12.1

I test two single room have better result than `VGGT`. While on matrix city dataset and two adjacent rooms work bad.Align result is abnormal. 
<img width="1773" height="277" alt="Image" src="https://github.com/user-attachments/assets/a98a12b5-19eb-48d6-8e05-198decbeb2b9" />

<img width="1631" height="594" alt="Image" src="https://github.com/user-attachments/assets/4c0a92f4-1e74-46f9-83c7-7b7575f8656a" />

<img width="1493" height="985" alt="Image" src="https://github.com/user-attachments/assets/4a4d8b89-1472-47d0-a54e-fdca9d83bd22" />

<img width="1624" height="494" alt="Image" src="https://github.com/user-attachments/assets/9250fd03-475d-4e9b-ac61-8ce62b764066" />
